### ISSUE TITLE
Code fix for `NUnit2005` tries to use `Is.Empty` constraint on incompatible types.

### ISSUE THREAD

**Issue by philipp-naused** [nunit/nunit.analyzers#952]
The `Is.Empty` constraint only works on certain types like `string`, `Guid`, `IEnumerable`, ect.
But the code fix for `NUnit2005` will still try to use it if the original code compared against a field called `Empty`, even if the type is not compatible.

Example:
```cs
using NUnit.Framework;
using NUnit.Framework.Legacy;

public enum Range
{
    Empty,
    Full
}

public class Test
{
    [Test]
    public void SampleTest()
    {
        var rangeValue = Range.Empty;
        ClassicAssert.AreEqual(Range.Empty, rangeValue);
    }
}
```
The code fix (e.g., `dotnet format --diagnostics NUnit2005`) will replace the assertion with:
```cs
Assert.That(rangeValue, Is.Empty);
```

This test will then fail with:
```
System.ArgumentException : The actual value must be not-null, a string, Guid, have an int Count property, IEnumerable or DirectoryInfo. The value passed was of type Range.
Parameter name: actual
  at NUnit.Framework.Constraints.EmptyConstraint.ApplyTo[TActual](TActual actual)
  at NUnit.Framework.Assert.That[TActual](TActual actual, IResolveConstraint expression, NUnitString message, String actualExpression, String constraintExpression)
  at Test.SampleTest()
```

Version: 4.11.2

**Comment by manfred-brands** [comment]
Thanks for the report, I will fix it tomorrow.

**Comment by manfred-brands** [comment]
This is caused because the code fixer tries to find cases of:  `XXX.Empty` to cover `String.Empty` and `ImmutableXXX<T>.Empty`
But it catches any member called Empty, including your two examples.
This is was intended for other collection types that follow the pattern and add an static `Empty` property.

The problem is that a CodeFixer only uses text, it doesn't have access to symbols so doesn't know any types.
It cannot check that `Empty` returns an instance of the class it is a member of.

So I need to either only detect explicit cases or pass information from the analyzer.
